### PR TITLE
Fix nameof operator

### LINKED ISSUE #11696: nameof(*) outputs "op_Multiply" but not "*"
nameof(\*) outputs "op_Multiply" but not "*"

**Repro steps**

```
Microsoft (R) F# Interactive version 11.4.2.0 for F# 5.0
Copyright (c) Microsoft Corporation. All Rights Reserved.

For help type #help;;

> nameof(*);;
val it : string = "op_Multiply"

> nameof( *);;
val it : string = "*"

> nameof( * );;
val it : string = "*"
```

**Expected behavior**

```
> nameof(*);;
val it : string = "*"
```

**Actual behavior**

```
> nameof(*);;
val it : string = "op_Multiply"
```

**Known workarounds**

```
> nameof( * );;
val it : string = "*"
```

**Related information**

Provide any related information (optional)::

* Operating system: Windows 10
* .NET Runtime kind (.NET Core, .NET Framework, Mono): .NET Framework FSI
* Editing Tools (e.g. Visual Studio Version, Visual Studio): VS16.10.2

### LINKED ISSUE #11697: nameof(..  ..) outputs "op_RangeStep" but not ".. .."
nameof(..  ..) outputs "op_RangeStep" but not ".. .."

**Repro steps**

```
Microsoft (R) F# Interactive version 11.4.2.0 for F# 5.0
Copyright (c) Microsoft Corporation. All Rights Reserved.

For help type #help;;

> nameof( .. .. );;
val it : string = ".. .."

> nameof( ..  .. );;
val it : string = "op_RangeStep"

> nameof( ..   .. );;
val it : string = "op_RangeStep"
```

**Expected behavior**

```
> nameof( .. .. );;
val it : string = ".. .."
```

**Actual behavior**

```
> nameof( ..  .. );;
val it : string = "op_RangeStep"
```

**Known workarounds**

```
> nameof( .. .. );;
val it : string = ".. .."
```

**Related information**

Provide any related information (optional)::

* Operating system: Windows 10
* .NET Runtime kind (.NET Core, .NET Framework, Mono): .NET Framework FSI
* Editing Tools (e.g. Visual Studio Version, Visual Studio): VS16.10.2

### PR DESCRIPTION
**Pull request by nojaf**
Fixes #11696 and fixes #11697

### CODE REVIEW

File: tests/FSharp.Compiler.ComponentTests/Language/NameofTests.fs [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+open FSharp.Test.Compiler
+
+module NameofTests =
+
+    [<Theory>]
+    [<InlineData("+")>]
+    [<InlineData("-")>]
+    [<InlineData("/")>]
+    [<InlineData("*")>]
+    [<InlineData(".. ..")>]
```
Comment (Happypig375) [inline tests/FSharp.Compiler.ComponentTests/Language/NameofTests.fs:15]:
```suggestion
    [<InlineData(".. ..")>]
    [<InlineData("..  ..")>]
    [<InlineData("..   ..")>]
    [<InlineData("<@@>")>]
    [<InlineData("<@ @>")>]
    [<InlineData("<@  @>")>]
    [<InlineData("<@   @>")>]
```

**Comment by Happypig375** [comment]
Should also add tests for
```
..  ..
..   ..
<@@>
<@ @>
<@  @>
<@   @>
```

**Comment by nojaf** [comment]
<pre>..               ..</pre> will be captured by the parser as <pre>.. ..</pre>, see 
https://github.com/dotnet/fsharp/blob/f89e6518d167705d5c0bae534dfb49c6a9a39f85/src/Compiler/pars.fsy#L5567-L5568

I don't see much value in restoring the exact number of spaces just to display it correctly in `nameof`.

**Comment by vzarytovskii** [comment]
> Should also add tests for
> 
> ```
> ..  ..
> ..   ..
> <@@>
> <@ @>
> <@  @>
> <@   @>
> <@@@@>
> <@@ @@>
> <@@  @@>
> <@@   @@>
> ```
> 
> and space on left, space on right, space on both sides

At first, I thought, it's some ASCII-art.

**Comment by Happypig375** [comment]
@nojaf The tests do not capture the # behaviour you described.

**Comment by nojaf** [comment]
https://fsprojects.github.io/fantomas-tools/#/ast?data=N4KABGBEAmCmBmBLAdrAzpAXFSAacUiaAYmolmPAIYA2as%2BEkAxgPZwWTJUC2sr8ABQA6YRFEBKADoAnKcm58BIseOHS5C3vyEAeAAL6AfBvmKdgg2GOmtSvfog3ZZ7cqtOTLuxYOHnmubuhtYBrvaWIaFegW4Ojk42kCAAvkA

Well no because expected and outcome might differ depending on spaces as I explained above.
Point it that this problem is fixed...

**Comment by Happypig375** [comment]
Well, have tests that capture the current behaviour so that it's maintained.

**Review by KevinRansom — APPROVED** [review]

**Review by 0101 — APPROVED** [review]

**Pull request merged**